### PR TITLE
Use the call stack to Record#new in the "Managed by kennel" line

### PR DESCRIPTION
**Pull request by zdrve**
near-duplicate of https://github.com/zdrve/kennel/pull/1 

(this one runs the tests; the other one has the right base branch)

### CODE REVIEW

File: lib/kennel/models/project.rb [hunk 1]
```diff
@@ -18,7 +18,10 @@ def self.file_location
       end
 
       def validated_parts
-        all = parts
+        # 'parts' called on same line as 'caller' for maximum cleanliness in strip_caller
+        here, all = [->() {caller}.call, parts]
+        all = [all] unless all.is_a?(Array)
```
Comment (grosser) [inline lib/kennel/models/project.rb:23]: can do this or `all ||= []`
```suggestion
        all = Array(all)
```

Comment (grosser) [inline lib/kennel/models/project.rb:23]:
```suggestion
        here, all = [->() {caller}.call, parts.to_a]
```

Comment (grosser) [reply lib/kennel/models/project.rb:23]: might be simpler to say `remove everything before __FILE__:__LINE__`

Comment (zdrve) [reply lib/kennel/models/project.rb:23]: The `all = [all]` was there for a purpose... however, this https://github.com/zendesk/kennel/pull/14227 (plus a follow-up grosser-kennel PR to fail if `parts` returns a non-array) is the better fix for that problem.

**Pull request closed without merging**